src/pytest_inmanta_lsm/remote_orchestrator.py

```diff
@@ -57,20 +60,34 @@ def __init__(
         :param settings: The inmanta environment settings that should be set on the remote orchestrator
         :param noclean: Option to indicate that after the run clean should not run. This exposes the attribute to other
                         fixtures.
+        :param ssl: Option to indicate whether SSL should be used or not. Defaults to false
+        :param token: Token used for authentication
+        :param ca_cert: Certificate used for authentication
         """
         self._env = environment
         self._host = host
         self._ssh_user = ssh_user
         self._ssh_port = ssh_port
         self._settings = settings
         self.noclean = noclean
+        self._ssl = ssl
+        self._token = token
+        self._ca_cert = ca_cert
 
         inmanta_config.Config.load_config()
         inmanta_config.Config.set("config", "environment", str(self._env))
-        inmanta_config.Config.set("compiler_rest_transport", "host", host)
-        inmanta_config.Config.set("compiler_rest_transport", "port", "8888")
-        inmanta_config.Config.set("client_rest_transport", "host", host)
-        inmanta_config.Config.set("client_rest_transport", "port", "8888")
+
+        for section in ["compiler_rest_transport", "client_rest_transport"]:
+            inmanta_config.Config.set(section, "host", host)
+            inmanta_config.Config.set(section, "port", "8888")
+
+            # Config for SSL and authentication:
+            if ssl:
+                inmanta_config.Config.set(section, "ssl", str(ssl))
+                if token:
+                    inmanta_config.Config.set(section, "token", token)
+                if ca_cert:
+                    inmanta_config.Config.set(section, "ssl_ca_cert_file", ca_cert)
```

Comment:
SSL and authentication can be enabled/disabled independently on the Inmanta server. SSL is enabled by setting the `server.ssl_cert_file` and the `server.ssl_key_file` config option. Authentication is enabled via the `server.auth` config option. As such, `pytest-inmanta-lsm` should be able to enable/disable SSL and authentication independently as well.

Reply:
I'm not sure I understand what you mean here.

Reply:
With the above-mention code it's not possible to make `pytest-inmanta-lsm` talk to an Inmanta orchestrator that has authentication enabled, but not SSL, because the `inmanta_config.Config.set(section, "token", token)` line is present under the `if ssl` statement. It would be good to decouple them. Does that make things more clear?

```suggestion
            if token:
                inmanta_config.Config.set(section, "token", token)
            if ssl:
                inmanta_config.Config.set(section, "ssl", str(ssl))
                if ca_cert:
                    inmanta_config.Config.set(section, "ssl_ca_cert_file", ca_cert)
```